documentation/general/program-scale-testing-requirements/hero-implementation-details.md

```diff
@@ -0,0 +1,261 @@
+
+| Metric                       | Baby Hero | % of HERO  |
+|------------------------------|-----------|------------|
+| VNIs                         | 32        | 3%         |
+| ENIs                         | 32        | 100%       |
+| NSGs per ENI                 | 10        | 100%       |
+| NSGs                         | 320       | 100%       |
+| ACL rules per NSG            |  1,000    | 100%       |
+| ACL rules                    |  320,000  | 100%       |
+| ACL prefixes per ACL rule    | 1         | 1%         |
+| ACL prefixes                 |  320,000  | 1%         |
+| ACL ports                    | NA        | 0%         |
+| VNETs (Outbound)             | 32        | 100%       |
+| Outbound routes per ENI      | 5K        |            |
+| Outbound routes              | 160K      | 5%         |
+| CA-to-PA mapping (Outbound)  | 80320     | 1%         |
+| Inbound Routes per ENI       | 1         | 0%         |
+| Inbound Routes               | 32        | 100%       |
+| PAs (Inbound PA Validation)  | 32        | 100%       |
+
+
+### In between scale
+
+Before the final solution is finished, we can add another checkpoint to collect further data if the Hero test scale numbers are not fulfilled.
+This will have custom scale values agreed upfront by all the parties and constitutes an intermediary point in the DASH development.
+Usually becomes irrelevant as soon as the Hero test scale is achieved.
+
+### Hero test
+
+The test exactly as specified in the DASH requirements.
+
+### Best case scenario
+
+If we can find a scenario where we obtain better performance numbers then the numbers previously obtained during (1ip, baby hero, hero test ...) this will be added as a new data point to the results.
+
+### Worst case scenario
+
+If we can find a scenario where we obtain lower performance numbers then the numbers previously obtained during (1ip, baby hero, hero test ...) this will be added as a new data point to the results.
+
+
+## Metrics
+
+Besides CPS we actually collect for all those different scale points multiple metrics.
+
+### PPS
+
+PPS (packets per second) is the first metric we collect. Usually, it is UDP packets are used
+Packets must be as small as possible, so we do not to hit link speed constraints.
+This test shows the fast path performance.
+
+### Latency
+
+Latency is the time it takes for a packet to go through the device under test.
+
+Latency value is most accurate when we have highest PPS, smallest packet, and zero packet loss. and is measured using IxNetwork and Novus card.
+
+When testing the smart switch we have to run a test to get the switch latency without running the traffic through the DPU and then get the total system latency with the understanding that each packets travel once through the NPU to reach the DPU, than it travel through teh DPU and once more it will travel through teh NPU after it leave the DPU. 
```

Comment:
s/teh/the/ (2 places)
s/travel/travels/

Reply:
done